src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -31,7 +34,7 @@ public String getHostaddress() {
         return hostaddress;
     }
 
-    public void getHostaddress(String hostaddress) {
+    public void setHostaddress(String hostaddress) {
```

Comment:
It sounds ExtensionSettings should be immutable, and if that is the case maybe there could be a builder to create objects of a specific configuration for testing?

Note, for tests this can be worked around by creating a mock `ExtensionSettings` object that returns the expected values.

Reply:
That's a good call out. Removed setters and found a way ObjectMapper can initialize in the constructor.